### PR TITLE
Fix HLO dump file names for PJRT

### PR DESCRIPTION
**Pull request by will-cromar**
`SaveTensorsGraphInfo` relied on a global save file path and ordinal (as an env var) and wrote all HLO dumps to the same file. Use thread local ordinal and save file path to support PJRT.

Added a PJRT multi-CPU test and tested manually with XRT.

### CODE REVIEW

File: test/pjrt/test_experimental_pjrt_multi_cpu.py [hunk 1]
```diff
@@ -108,6 +108,23 @@ def test_spawn(self, spawn):
...
+    xm.mark_step()
+    x.cpu()
+
+  def test_hlo_dump(self):
+    tmpdir = self.create_tempdir().full_path
+    pjrt._run_multiprocess(self._hlo_dump, tmpdir)
+
+    files = os.listdir(tmpdir)
+    for i in range(4):
+      self.assertIn(f'save.hlo.{i}', files)
```
Comment (vanbasten23) [inline test/pjrt/test_experimental_pjrt_multi_cpu.py:126]: do you need to clean up the hlo dump at the end of the test?

File: test/pjrt/test_experimental_pjrt_multi_cpu.py [hunk 2]
```diff
@@ -108,6 +108,23 @@ def test_spawn(self, spawn):
     indices = sorted(queue.get(block=False) for _ in range(queue.qsize()))
     self.assertListEqual(indices, list(range(4)))
 
+  @staticmethod
+  def _hlo_dump(tmpdir: str):
+    os.environ['XLA_SAVE_TENSORS_FMT'] = 'hlo'
+    os.environ['XLA_SAVE_TENSORS_FILE'] = os.path.join(tmpdir, 'save.hlo')
```
Comment (JackCaoG) [inline test/pjrt/test_experimental_pjrt_multi_cpu.py:114]: hmm, I actually don't know if `os.environ` is local to this function. Will `os.environ` got reset upon return of this function?

File: torch_xla/csrc/debug_util.cpp [hunk 3]
```diff
@@ -116,8 +116,8 @@ void DebugUtil::SaveTensorsGraphInfo(const char* name,
                                      absl::Span<const XLATensorPtr> tensors,
                                      const std::vector<size_t>* indices,
                                      GraphFormat format) {
-  static const std::string save_file =
-      xla::sys_util::GetEnvOrdinalPath("XLA_SAVE_TENSORS_FILE", "");
+  thread_local const std::string save_file = xla::sys_util::GetEnvOrdinalPath(
```
Comment (JackCaoG) [inline torch_xla/csrc/debug_util.cpp:119]: haha I need to google first to confirm that `thread_local` is implicitly static.

File: test/pjrt/test_experimental_pjrt_multi_cpu.py (hunk 2 above)
Comment (will-cromar) [reply test/pjrt/test_experimental_pjrt_multi_cpu.py:114]: It's not local, but we only change `os.environ` in the sub process, so it does get reset between tests

File: torch_xla/csrc/debug_util.cpp (hunk 3 above)
Comment (will-cromar) [reply torch_xla/csrc/debug_util.cpp:119]: Same! https://stackoverflow.com/a/22794640

File: test/pjrt/test_experimental_pjrt_multi_cpu.py (hunk 1 above)
Comment (will-cromar) [reply test/pjrt/test_experimental_pjrt_multi_cpu.py:126]: No. We got the temp dir from the test class, which cleans up temp files at the end

File: test/pjrt/test_experimental_pjrt_multi_cpu.py [hunk 4]
```diff
@@ -108,6 +108,23 @@ def test_spawn(self, spawn):
...
+
+    x = torch.randn((3, 3), device=xm.xla_device())
+    xm.mark_step()
+    x.cpu()
+
+  def test_hlo_dump(self):
```
Comment (JackCaoG) [inline test/pjrt/test_experimental_pjrt_multi_cpu.py:120]: can we add the same test under maybe `test_operations`? I want to make sure this change does not break non-pjrt work loads. If it is too complicated we can do it in a separate pr.

Comment (will-cromar) [reply test/pjrt/test_experimental_pjrt_multi_cpu.py:120]: This is actually already tested on XRT under this test case: https://github.com/pytorch/xla/blob/master/test/dynamo/test_dynamo_graph_dump.py

I know that because it caught a bug in the name of my files :smile:

Comment (JackCaoG) [reply test/pjrt/test_experimental_pjrt_multi_cpu.py:120]: lol OK, I forgot I added that test.

**Review by JackCaoG — APPROVED** [review]
